src/dtx/dtx_rpc.c

```diff
@@ -367,7 +367,11 @@ dtx_cf_rec_update(struct btr_instance *tins, struct btr_record *rec,
 
 	drr = (struct dtx_req_rec *)umem_off2ptr(&tins->ti_umm, rec->rec_off);
 	dcrb = (struct dtx_cf_rec_bundle *)val->iov_buf;
-	drr->drr_dti[drr->drr_count++] = *dcrb->dcrb_dti;
+	if (!daos_dti_equal(&drr->drr_dti[drr->drr_count], dcrb->dcrb_dti)) {
```

Comment:
How does this work?   It's only checking the entry at drr->drr_count which wouldn't be initialized because it's the "next" slot, no?

Shouldn't it check all of the entries?

Reply:
You are right, the one to be checked is "drr->drr_dti[drr->drr_count - 1]"

Reply:
so are the equivalent ones guaranteed to be contiguous ?

Reply:
Yes, it is guaranteed. Because the DTX ID is unique. The different shards with the same DTX ID must belong to the same TX, they will be parsed contiguously.